3.1.0 pre release build failure on alpinelinux

I started packaging the 3.1.0 pre release on alpine linux [1] and the build is failing on all arch with the same error

```
In file included from /usr/include/fortify/string.h:26,
                 from /home/simon/aports/testing/sdl3/src/SDL3-3.1.0/include/SDL3/SDL_stdinc.h:38,
                 from /home/simon/aports/testing/sdl3/src/SDL3-3.1.0/include/SDL3/SDL.h:32,
                 from /home/simon/aports/testing/sdl3/src/SDL3-3.1.0/test/testevdev.c:14:
/usr/include/fortify/stdlib.h:42:1: error: 'realpath' undeclared here (not in a function)
   42 | _FORTIFY_FN(realpath) char *realpath(const char *__p, char *__r)
      | ^~~~~~~~~~~
In file included from /home/simon/aports/testing/sdl3/src/SDL3-3.1.0/test/../src/SDL_internal.h:90,
                 from /home/simon/aports/testing/sdl3/src/SDL3-3.1.0/test/testevdev.c:26:
/usr/include/fortify/stdlib.h: In function 'realpath':
/usr/include/fortify/stdlib.h:45:2: error: #error PATH_MAX unset. A fortified realpath will not work.
   45 | #error PATH_MAX unset. A fortified realpath will not work.
      |  ^~~~~
```

[1]: https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/62841

--- a/test/testevdev.c
+++ b/test/testevdev.c
@@ -11,11 +11,10 @@
   freely.
 */
 
-#include <SDL3/SDL.h>
-#include <SDL3/SDL_main.h>
-#include <SDL3/SDL_test.h>
+/* Hack #1: avoid inclusion of SDL_main.h by SDL_internal.h */
+#define SDL_main_h_
 
-/* Hack to avoid dynapi renaming */
+/* Hack #2: avoid dynapi renaming (must be done before #include <SDL3/SDL.h>) */
 #include "../src/dynapi/SDL_dynapi.h"
 #ifdef SDL_DYNAMIC_API
 #undef SDL_DYNAMIC_API
@@ -25,6 +24,18 @@
 #ifndef NO_BUILD_CONFIG
 #include "../src/SDL_internal.h"
 #endif
+
+/* Hack #3: undo Hack #1 */
+#ifdef SDL_main_h_
+#undef SDL_main_h_
+#endif
+#ifdef SDL_MAIN_NOIMPL
+#undef SDL_MAIN_NOIMPL
+#endif
+
+#include <SDL3/SDL.h>
+#include <SDL3/SDL_main.h>
+#include <SDL3/SDL_test.h>
 
 #include <stdio.h>
 #include <string.h>

